src/pages/home/report/ReportActionCompose.js

```diff
@@ -317,16 +314,24 @@ class ReportActionCompose extends React.Component {
      * @param {String} emoji
      */
     addEmojiToTextBox(emoji) {
-        const emojiWithSpace = `${emoji} `;
-        const newComment = this.comment.slice(0, this.state.selection.start)
-            + emojiWithSpace + this.comment.slice(this.state.selection.end, this.comment.length);
-        this.setState(prevState => ({
-            selection: {
-                start: prevState.selection.start + emojiWithSpace.length,
-                end: prevState.selection.start + emojiWithSpace.length,
-            },
-        }));
-        this.updateComment(newComment);
+        const hasRangeSelected = this.selection.start !== this.selection.end;
+        if (Platform.OS === 'android' && hasRangeSelected) {
```

Comment:
It seems like we need to spell this out more explicitly in our style guidelines somewhere (or maybe with an eslint rule), but we have a pretty strict ban on using `Platform.OS`, `Platform.select`, or other `Platform` APIs in this repo. Instead we always create a small lib for the specific function and use platform-specific file extensions.

This is the closest we have to explaining this, but it could be made more explicit: https://github.com/Expensify/App/blob/main/README.md#platform-specific-file-extensions

Reply:
Agree it seems like a RN bug. If you create an issue in the RN repo with a minimum reproduction we will sponsor a fix.

Reply:
+1. let's refactor this in any case. If you want to fix this on RN as Rory said, we can consider holding this PR for the fix. If this is urgent (The internal team will know), then this is fine until a real fix is in place.

Reply:
Alright cool. Yeah, i identified a bunch of bugs in relation to selection. I would like to create reproductions for all of them, then share them with you, and then we can discuss sponsoring them?

I can't really estimate how long it would took me to fix those bugs, as I am not too familiar with the native code of the TextInput components yet. So i think it would make sense to merge that already and at some point remove the platform dependent code (just pushed that)

Reply:
yeah, agree with not holding this PR and just making a follow-up issue to remove this platform dependent code after the upstream fixes are merged.

Reply:
I just tried to reproduce the issues, but it seems to be related to a lag between the native side and JS. When we add an emoji, the JS thread basically gets frozen; thus, it's not catching events from the native side, leading to strange results (i wasn't really able to reproduce the issues in isolation.)
I presume that this could be gone when we have fabric.

(In general, I would like to make the whole "composer experience" more performant. The "picking emojis" and "sending comments" flow, is super slow on my devices and imo is really a bummer in UX. Which would presumably fix the issues I am seeing on android with the selection as there won't be any lag anymore).

So the only issue that's really reproducible is the missing context menu when using the `selection` prop (https://github.com/Expensify/App/issues/8349), however, I am not sure if it's still worth fixing that on the native side when it seems to be resolved with fabric?

Reply:
I wasn't 100% right on this one, I was able to reproduce the issue:

https://github.com/facebook/react-native/issues/35005